llvm/lib/Target/ARM/Thumb1InstrInfo.cpp

```diff
@@ -135,14 +135,15 @@ void Thumb1InstrInfo::loadRegFromStackSlot(MachineBasicBlock &MBB,
 void Thumb1InstrInfo::expandLoadStackGuard(
     MachineBasicBlock::iterator MI) const {
   MachineFunction &MF = *MI->getParent()->getParent();
-  const TargetMachine &TM = MF.getTarget();
   const ARMSubtarget &ST = MF.getSubtarget<ARMSubtarget>();
+  const GlobalValue *GV =
+      cast<GlobalValue>((*MI->memoperands_begin())->getValue());
```

Comment:
`const auto *GV = ...`

https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

(this can be fixed in llvm/lib/Target/ARM/Thumb2InstrInfo.cpp L264 as well)

Reply:
Thanks for the suggestion. Done